clippy_dev/src/lintcheck.rs

```diff
@@ -116,9 +115,9 @@ impl CrateSource {
                 // url to download the crate from crates.io
                 let url = format!("https://crates.io/api/v1/crates/{}/{}/download", name, version);
                 println!("Downloading and extracting {} {} from {}", name, version, url);
-                let _ = std::fs::create_dir("target/lintcheck/");
-                let _ = std::fs::create_dir(&krate_download_dir);
-                let _ = std::fs::create_dir(&extract_dir);
+                std::fs::create_dir("target/lintcheck/").expect("cannot create lintcheck target dir");
+                std::fs::create_dir(&krate_download_dir).expect("cannot create crate download dir");
+                std::fs::create_dir(&extract_dir).expect("cannot create crate extraction dir");
 
```

Comment:
This panics now because the directory may already exist (previously we just ignored the error)
I think we can work around by using `std::fs::create_dir_all`